tools/vegan_rarefaction/vegan_rarefaction.xml

```diff
@@ -0,0 +1,117 @@
+##library(Hmisc)
+##minor.tick(nx=2, ny=2, tick.ratio=0.5)
+invisible(dev.off())
+
+    ]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_abundance" type="data" format="tabular" label="File with abundance values for community" help="Rows are samples; columns are species/phyla/community classifier"/>
+        
+        <!-- spec column label?, counts (spec in rows) single select
+        treatment columns multi select, 1+
+        -->
+        
+        <param name="species_column" label="Group name column" type="data_column" data_ref="input_abundance" value="6" help="Species, phylum, etc"/>
+        <param name="sample_columns" label="Sample count columns" type="data_column" multiple="True" value="2" data_ref="input_abundance" help="Select each column that contains counts"/>
+        <!-- <param name="swap_columns_rows" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Swap rows and columns" help="Rows are species/phyla/community classifier; columns are samples"/> -->
+        <param name="sample_size" type="integer" value="" min="1" optional="True" label="Subsample size for rarefying community" help="leave empty to subsample by community size for the smallest sample"/>
+        <!-- <param name="estimate_standard_error" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="False" label="Estimate standard errors"/> -->
+        <!-- <param name="margin" type="integer" value="1" label="Margin for which the index is computed" /> -->
+        
+        <param name="step_size" type="integer" value="1" min="1" label="Step size for sample sizes"/>
+        <param name="xlab" type="text" value="Sample Size" label="X-axis label"/>
+        <param name="ylab" type="text" value="Species" label="Y-axis label"/>
+        <param name="label" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="True" label="Label rarefaction curves by rownames of X"/>
+
+        
+    </inputs>
+    <outputs>
+        <data format="tabular" name="output_species_count" label="${tool.name} on ${on_string} (number of species)"/>
+        <data format="tabular" name="output_species_frequency" label="${tool.name} on ${on_string} (frequency of species)"/>
+        <data format="tabular" name="output_richness" label="${tool.name} on ${on_string} (estimated richness)"/>
+        <data format="tabular" name="output_species_probabilities" label="${tool.name} on ${on_string} (species probabilities)"/>
+        <data format="tabular" name="output_slope" label="${tool.name} on ${on_string} (slope of curve)"/>
+        <data format="txt" name="vegan_rarefaction_script_output" label="${tool.name} on ${on_string} (Rscript)" />
+        <data format="png" name="output_plot" label="${tool.name} on ${on_string} (plot)" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="fastq_input|fastq_input_selector" value="single" />
```

Comment:
It appears as though this may be copy/pasted from @blankenberg's cd-hit-dup wrapper.